src/mcts/search.cc

```diff
@@ -1444,16 +1445,16 @@ void SearchWorker::FetchSingleNodeResult(NodeToProcess* node_to_process,
     // Note that we want to calculate (exp(p-max_p))^(1/T) = exp((p-max_p)/T).
     p = FastExp((p - max_p) / params_.GetPolicySoftmaxTemp());
 
-    // Note that p now lies in [0, 1], so it is safe to store it in compressed
-    // format. Normalization happens later.
-    edge.edge()->SetP(p);
-    // Edge::SetP does some rounding, so only add to the total after rounding.
-    total += edge.edge()->GetP();
+    intermediate_[counter++] = p;
+    total += p;
   }
+  counter = 0;
   // Normalize P values to add up to 1.0.
   if (total > 0.0f) {
     const float scale = 1.0f / total;
-    for (auto edge : node->Edges()) edge.edge()->SetP(edge.GetP() * scale);
+    for (auto edge : node->Edges()) edge.edge()->SetP(intermediate_[counter++] * scale);
+  } else {
+    for (auto edge : node->Edges()) edge.edge()->SetP(intermediate_[counter++]);
   }
```

Comment:
nit: I don't know if my suggestion is acceptable style, but as mentioned earlier, the expected common case is `total > 0` so slowing down the exceptional case with "unnecessary" multiply might be acceptable…
```c
const auto scale = total > 0.0f ? 1.0f / total : 1.0f;
for (auto edge : node->Edges()) edge.edge()->SetP(intermediate[counter++] * scale);
```

Reply:
Thanks, this is a lot cleaner.
```
===========================
Total time (ms) : 340545
Nodes searched  : 70147040
Nodes/second    : 205984
```